app/controllers/submissions_controller.rb

```diff
@@ -2,7 +2,7 @@ class SubmissionsController < ApplicationController
   before_action :authenticate_user!, only: [:create, :create_comment]
   before_action :authenticate_to_submit, only: :new
 
-  helper_method :posts, :post, :new_post, :new_comment
+  helper_method :post, :posts, :new_submission, :new_comment, :parent_id
 
   def new
```

Comment:
FWIW, you can add `@submission = Submission.new` in this method, replace all instances of `new_submission` with `post` in the views and drop the `new_submission` method in this controller. I'd also recommend aligning the instance variable (`@submission`) with the reader method name (`post`).

Reply:
Ok, thanks. I'll try that. If I can clarify these things, then I could avoid splitting the model right now.